### PR TITLE
HIPP-546: IDMS stub authenticates incoming requests

### PR DESCRIPTION
**Pull request by PaulCDurham**
https://jira.tools.tax.service.gov.uk/browse/HIPP-546

### CODE REVIEW

File: test/uk/gov/hmrc/identitymanagementservicestubs/controllers/ClientsControllerSpec.scala [hunk 1]
```diff
@@ -60,12 +62,24 @@ class ClientsControllerSpec extends AnyFreeSpec with Matchers with MockitoSugar
...
         val request = FakeRequest(GET, routes.ClientsController.getClientSecret(clientId).url)
+          .withHeaders((AUTHORIZATION, "Basic aWRtcy1zdHViLWNsaWVudC1pZDppZG1zLXN0dWItc2VjcmV0"))
+
         when(fixture.idmsService.getSecret(clientId)).thenReturn(Future.successful(None))
         val result = route(fixture.application, request).value
         status(result) mustBe Status.NOT_FOUND
       }
     }
 
+    "must return 401 Unauthorized if no valid credentials are presented" in {
```
Comment (llmikeyj) [inline test/uk/gov/hmrc/identitymanagementservicestubs/controllers/ClientsControllerSpec.scala:73]: Minor point - this spec tests missing auth header, not the validity of the creds. Should we have at least one spec where we test an invalid but present basic auth header?

Comment (PaulCDurham) [reply test/uk/gov/hmrc/identitymanagementservicestubs/controllers/ClientsControllerSpec.scala:73]: I think that's the responsibility of AuthenticatorSpec. In this class I'm just trying to show that authentication is being applied.

Comment (llmikeyj) [reply test/uk/gov/hmrc/identitymanagementservicestubs/controllers/ClientsControllerSpec.scala:73]: Ok

**Review by llmikeyj — APPROVED** [review]
LGTM.

**Pull request merged**